test/xUnit/csharp/test_PowerShell.cs

```diff
@@ -0,0 +1,167 @@
+            }
+        }
+
+        [Fact]
+        public static async Task TestPowerShellInvokeAsyncWithInput()
+        {
+            using (var ps = PowerShell.Create())
+            {
+                ps.AddCommand("Get-Command");
+
+                var results = await ps.InvokeAsync(new PSDataCollection<string>(new[] { "Get-Process" }));
+
+                Assert.Single(results);
+                Assert.IsType<CmdletInfo>(results[0]?.BaseObject);
+                Assert.Equal("Get-Process", ((CmdletInfo)results[0].BaseObject).Name);
+            }
+        }
+
+        [Fact]
+        public static async Task TestPowerShellInvokeAsyncWithInputAndOutput()
+        {
+            using (var ps = PowerShell.Create())
+            {
+                ps.AddCommand("Get-Command");
+
+                var results = new PSDataCollection<CmdletInfo>();
+                await ps.InvokeAsync(new PSDataCollection<string>(new[] { "Get-Process" }), results);
+
+                Assert.Single(results);
+                Assert.IsType<CmdletInfo>(results[0]);
+                Assert.Equal("Get-Process", results[0].Name);
+            }
+        }
+
+        [Fact]
+        public static async Task TestPowerShellInvokeAsyncWithErrorInScript()
+        {
+            using (var ps = PowerShell.Create())
+            {
+                ps.AddScript(@"
+try {
+    Get-Process -Invalid 42 -ErrorAction Stop
+} catch {
+    throw
+}
+");
+                var results = await ps.InvokeAsync();
+
+                Assert.True(results.IsFaulted);
+                Assert.IsType<AggregateException>(results.Exception);
+                Assert.IsType<ParameterBindingException>(results.Exception.InnerException);
+                Assert.Equal(results.Exception.InnerException.CommandInvocation.InvocationName, "Get-Process");
+                Assert.Equal(results.Exception.InnerException.ParameterName, "Invalid");
+                Assert.Equal(results.Exception.InnerException.ErrorId, "NamedParameterNotFound");
+            }
+        }
+
+        [Fact]
+        public static async Task TestPowerShellInvokeAsyncInUnopenedRunspace()
+        {
+            using (var rs = RunspaceFactory.CreateRunspace())
+            using (var ps = PowerShell.Create(rs))
+            {
+                var results = await ps.AddScript(@"@(1..10).foreach{Start-Sleep -Milliseconds 500}")
+                                      .InvokeAsync();
+
+                Assert.True(results.IsFaulted);
+                Assert.IsType<AggregateException>(results.Exception);
+                Assert.IsType<InvalidRunspaceStateException>(results.Exception.InnerException);
+                Assert.Equal(results.Exception.InnerException.CurrentState, RunspaceState.BeforeOpen);
+                Assert.Equal(results.Exception.InnerException.ExpectedState, RunspaceState.Opened);
+            }
+        }
+
+        [Fact]
+        public static async Task TestPowerShellInvokeAsyncInBusyRunspace()
```

Comment:
I don't think we need this test again in C#. We are using the APIs directly even when testing in Pester.

Reply:
That begs the question then what tests do we need in C#? @BrucePay suggested having tests in both C# and PowerShell, when I had originally just started writing the tests in C#. Then I flushed out all of the PowerShell tests and the tests I'm writing in C# right now are pretty much mirrors of the tests I had written in PowerShell, which just use the API directly. Now that the PowerShell tests are written, do we need a C# test harness for this PR at all?

Reply:
I'm fine to not have the xUnit tests for this PR. The Pester tests are using the APIs directly anyways, except that they are not using the APIs in async/await manner. But that's a given in C# anyways.

Reply:
Removed.